### DISCUSSION TITLE
PhantomCamera3D is left further and further behind when attached to an accelerating RigidBody3D

### DISCUSSION THREAD

**Discussion by Employee2393** [ramokz/phantom-camera#394] [Question / Help Wanted] [answered]
Hi,

I have a scene that looks something like this:

* RigidBody3D
  * [Smoothing](https://github.com/lawnjelly/smoothing-addon) (without Target)
     * MeshInstance3D
         * PhantomCamera3D

(Also there is a Camera3D with PhantomCameraHost added to the scene root.)

The camera is configured like this: no follow mode, no follow target, no look at, it should work as a first person camera in this instance.

The RigidBody3D is accelerating in the scene, and the camera is smoothly left further and further behind.

I've noticed that there is always a small "delay"/interpolation/tweening when the velocity (magnitude or direction) changes.

Is there a way to set a limit for this behavior?

I don't want my camera to be left outside of the model or leave it behind too much.

**Comment by ramokz** [comment, marked as answer]
This is something I have seen since the early days of the project.
The added physics interpolation in Godot 4.3 addressed the issue for 2D cameras, but can see it is missing a slight tweak to work when there is no `Follow Mode` / `Follow Target` being set, and if you add it as a child of a physics node. So will add that in a near future patch release.

I am going to assume that it will be resolved for 3D cameras as well in Godot 4.4 when 3D physics interpolation gets added.

Added two 2D video examples below, one showing how it appears if the camera isn't applying physics interpolation and the other when it is being applied. In both examples, no `Follow Mode` is being set and the `PCam2D` is just a child of the `CharacterController2D`.

## Without 2D Physics Interpolation

https://github.com/user-attachments/assets/3d236fb4-fc55-4e94-bd6d-2c98bb5bc6b6

---

## With 2D Physics Interpolation

https://github.com/user-attachments/assets/02d75c66-bf7f-4494-ab2c-41e580d88cca
